src/main/java/uk/ac/ebi/tsc/portal/api/team/controller/TeamResource.java

```diff
@@ -115,5 +122,21 @@ public void setManagerUserNames(List<String> managerUserNames) {
 		this.managerUserNames = managerUserNames;
 	}
 
-	
+	public Collection<String> getTeamContactEmails() {
+		return teamContactEmails;
+	}
+
+	public void setTeamContactEmails(Collection<String> teamContactEmails) {
+		this.teamContactEmails = teamContactEmails;
+	}
+
+	private boolean isOwner(Team team) {
+		uk.ac.ebi.tsc.aap.client.model.User user = (uk.ac.ebi.tsc.aap.client.model.User) SecurityContextHolder.getContext().getAuthentication().getDetails();
+			if (user.getUserReference().equals(team.getAccount().getUsername())) {
+			return true;
+		}
+		return false;
+	}
```

Comment:
Sorry - but this is a terrible idea to tie a plain POJO with SecurityContext.
Besides (minor in comparison with the main problem), if you call a method of a class TeamResource `isOwner` it suggests that this method is checking, if the TeamResource instance is an owner of a team.
Finally, you made `isOwner` an instance method and it does not manipulate any data that belongs to TeamResource (no making it a static method of this class is also not a good idea).

Reply:
Has been implemented in a different way now.